### DISCUSSION TITLE
Plans for Matter (over Thread)

### DISCUSSION THREAD

**Discussion by joe2xyz** [Koenkk/zigbee2mqtt#21368] [General]
Hi folks,

Thank you @Koenkk and all maintainers and developers for this amaaazing project!

Are there any plans for Matter-over-Thread?

I'm not asking about supporting dongles that can do both Zigbee and Matter and I'm not asking about interop between Zigbee and Matter. I'm asking if you're considering supporting Matter only USB dongles and Matter-over-Thread devices?

I love how Zigbee2MQTT works and I'd prefer one day (when I get some Matter) to use it with Home Assistant instead of using HA builtin support for Matter.

If yes, is it going to be a new project (Matter2MQTT) or the same project that will "simply" work with Zigbee / Matter devices based on the connected USB dongle?

Thank you.

Joe

**Comment by Koenkk** [comment]
I have no plans for Matter2MQTT but maybe Matter2Zigbee using [matter.js](https://github.com/project-chip/matter.js/) (once matter.js project is ready for it)

**Comment by joe2xyz** [comment]
Thank you for the answer @Koenkk ;-)

Reading doc of matter.js and between lines in your answer:

1. You don't want to create a separate project for Matter (Matter2MQTT)
2. Matter.js might be a part of zigbee2mqtt and therefore the existing project (zigbee2mqtt) might get support for Matter. Possibly with two dongles (Zigbee and Matter) or one that supports multi-protocol.

Am I reading that correctly?

**Reply by sjorge** [reply]
No, it would not be a matter (over thread) <-> mqtt bridge but  zigbee <-> matter bridge.  Were z2m would explose matter compatible devices.

**Comment by joe2xyz** [comment]
In other words, old (Zigbee) devices would look like they can speak Matter.

That's sad news - I was hoping that you (Z2m developers and maintainers) would try to create a border router.

That means that the only open-source border router will be the Home-assistant implementation.

Of course, it is better to have one than none. But for example one thing I currently enjoy with Z2m is that I can run two separate networks (by running 2 instances of Z2m). There is no way I'll be able to run two Matter networks with HA since ZHA cannot have two coordinators and therefore I doubt they'll ever implement capability to run multiple border router instances.

**Reply by Koenkk** [reply]
> In other words, old (Zigbee) devices would look like they can speak Matter.

Exactly!

I would love to start a Matter2MQTT project but I simply don't have the time to do so 😄

**Reply by stefa168** [reply]
In the following months I might have more free time; it could be an interesting project to work on 😃

**Comment by joe2xyz** [comment]
Completely understand! Thank you!

Maybe somebody else will run with this idea ;-). It'd be a great duo - M2M and Z2M for example sharing the some code and DB of firmware updates.

**Comment by dersch81** [comment]
I would like to understand this more in detail. Isn't it possible to just integrate the Matter Protocol into z2m? As i understand some coordinators are able to speak both at the same time and it is the same frequency. What is the difficulty to expand z2m to it?

**Comment by chrigu1981** [comment]
I love this idea. @Koenkk are there still no plans to integrate Matter into Z2M?

**Reply by Koenkk** [reply]
No, Matter and Zigbee are too different therefore it doesn't make sense to do it. Matter2Mqtt is a project on it's own (and I don't have the time for it)
